### PR TITLE
Typecheck DINO loss

### PR DESCRIPTION
**Pull request by philippmwirth**
# Typecheck DINO loss

Typecheck DINO loss and tests. I had to refactor the tests a bit but the cases should not have changed.

### CODE REVIEW

File: tests/loss/test_dino_loss.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    """
+    torch.manual_seed(seed)
+    out = []
+    for _ in range(n_views):
+        views = [torch.rand(output_dim) for _ in range(batch_size)]
+        out.append(torch.stack(views))
+    return out
+
+
+def _run_test(
```
Comment (MalteEbner) [inline tests/loss/test_dino_loss.py:170]: I would explain better what this function does, at least with the name.
```suggestion
def _assert_dino_loss_equal_to_original(
```

**Review by MalteEbner — COMMENTED** [review]
One open question.

File: lightly/loss/dino_loss.py [hunk 2]
```diff
@@ -161,6 +164,6 @@ def update_center(self, teacher_out: Tensor) -> None:
         batch_center = self._center_fn(x=teacher_out, dim=(0, 1))
 
         # Update the center with a moving average
-        self.center = center.center_momentum(
+        self.center.data = center.center_momentum(
```
[the remark this answers is not in the document]
Comment (philippmwirth) [reply lightly/loss/dino_loss.py:167]: In the constructor, see the line:
```python
self.register_buffer("center", torch.zeros(1, 1, output_dim))
```

File: tests/loss/test_dino_loss.py (hunk 1 above)
Comment (philippmwirth) [reply tests/loss/test_dino_loss.py:170]: I think you were the original author and the function was named `test` but ok let me rename it.

Comment (MalteEbner) [reply tests/loss/test_dino_loss.py]: It was named `def _assert_dino_loss_equal_to_original(` in the original test, you changed it when copy-pasting it.

Comment (MalteEbner) [reply tests/loss/test_dino_loss.py]: Feel free to resolve.

**Review by MalteEbner — APPROVED** [review]
LGTM
